File: app/src/main/java/com/elishaazaria/sayboard/preferences/MyPreferences.java
package com.elishaazaria.sayboard.preferences;

import android.annotation.SuppressLint;
import android.content.Context;
import android.content.SharedPreferences;

import com.elishaazaria.sayboard.AppCtx;
import com.elishaazaria.sayboard.R;

public class MyPreferences {
    private static SharedPreferences sharedPref;

    public static SharedPreferences getSharedPref() {
        if (sharedPref == null) {
            sharedPref = AppCtx.getAppCtx().getSharedPreferences(
                    AppCtx.getStringRes(R.string.main_shared_pref), Context.MODE_PRIVATE);
        }
        return sharedPref;
    }
}

File: app/src/main/java/com/elishaazaria/sayboard/preferences/MyPreferences.kt
package com.elishaazaria.sayboard.preferences

import android.content.Context
import android.content.SharedPreferences
import com.elishaazaria.sayboard.AppCtx.appCtx
import com.elishaazaria.sayboard.AppCtx.getStringRes
import com.elishaazaria.sayboard.R

object MyPreferences {
    private var mySharedPref: SharedPreferences? = null

    val sharedPref: SharedPreferences
        get() {
            if (mySharedPref == null) {
                mySharedPref = appCtx!!.getSharedPreferences(
                    getStringRes(R.string.main_shared_pref), Context.MODE_PRIVATE
                )
            }
            return mySharedPref!!
        }
}